### ISSUE TITLE
na_ontap_s3_service module or na_ontap_svm module needs to return access key and secret key

### ISSUE THREAD

**Issue by vhulikal** [ansible-collections/netapp.ontap#193]
### Summary

I am using netapp.ontap galaxy collection 22.7.0. When I call na_ontap_s3_users, it returns me the access key and the secret key. But when I call na_ontap_s3_services, it will not return the access key or the secret key; I need these values to configure S3 browser. I think, this access key and the secret key belon to sm_s3_user. I observe the same behavior when I call na_ontap_svm, when I am using theS3 protocol. I think, na_ontap_svm module should also return the access key and the secret key for the sm_s3_user.

### Component Name

netapp.ontap galaxy collection 22.7.0

### Ansible Version

```console
[root@centos1 ansible-workshop]# ansible --version
ansible [core 2.15.3]
  config file = /root/ansible-workshop/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.11/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.11.4 (main, Sep  7 2023, 15:21:38) [GCC 8.5.0 20210514 (Red Hat 8.5.0-4)] (/usr/local/bin/python3.11)
  jinja version = 3.1.2
  libyaml = True
```

### ONTAP Collection Version

```console
$ ansible-galaxy collection list
netapp.ontap                  22.7.0
```

### ONTAP Version

```console
cluster1::> version
NetApp Release 9.13.1: Mon Jun 19 07:42:08 UTC 2023
```

### Playbook

```console
tasks:
  - name: Create an s3 service on primary SVM
    na_ontap_s3_services:
      state:            present
      name:             "s3_server"
      vserver:          "{{ PRI_SVM }}"
      comment:          not enabled
      enabled:          false
      certificate_name: nas_svm_1791794040043EE3
      <<: *input
```

### Steps to Reproduce

#!/usr/bin/env ansible-playbook

################################################################################
#
# Title:        44_create_s3_service.yml
# Author:       Vish Hulikal
# Date:         2023-09-14
# Description:  Create an S3 service
#               with Ansible module
# Procedure to get the certificate name:
# 1. Go to System Manager -> Cluster -> Settings-> Certificates -> Client/Server Certificates
# 2. Check the certificate for the nas_svm, which you just created by running the playbook
#               41_create_pri_svm.yml
# 3. Note it down: nas_svm_1791794040043EE
#
# Modules:      na_ontap_svm
#
# URLs:         http://docs.netapp.com/ontap-9/index.jsp
#               https://galaxy.ansible.com/netapp/ontap
#
# Built-in help:
# ansible-doc netapp.ontap.na_ontap_svm
#
################################################################################

- hosts: localhost
  gather_facts: false
  vars:
    input: &input
      hostname:         "{{ PRI_CLU }}"
      username:         "{{ PRI_CLU_USER }}"
      password:         "{{ PRI_CLU_PASS }}"
      https:            true
      validate_certs:   false
      use_rest:         Always
  vars_files:
    - global_nas.vars
  collections:
    - netapp.ontap

  tasks:
  - name: Create an s3 service on primary SVM
    na_ontap_s3_services:
      state:            present
      name:             "s3_server"
      vserver:          "{{ PRI_SVM }}"
      comment:          not enabled
      enabled:          false
      certificate_name: nas_svm_1791794040043EE3
      <<: *input

### Expected Results

I expected it to return access key and the secret key and it did not:
-----------------------------------------
#!/usr/bin/env ansible-playbook
- hosts: localhost
  gather_facts: false
  vars:
    input: &input
      hostname:         "{{ PRI_CLU }}"
      username:         "{{ PRI_CLU_USER }}"
      password:         "{{ PRI_CLU_PASS }}"
      https:            true
      validate_certs:   false
      use_rest:         Always
  vars_files:
    - global_nas.vars
  collections:
    - netapp.ontap

  tasks:
  - name: Create an s3 service on primary SVM
    na_ontap_s3_services:
      state:            present
      name:             "s3_server"
      vserver:          "{{ PRI_SVM }}"
      comment:          enabled
      enabled:          true
      certificate_name: nas_svm_179219CA97BAD25E
      <<: *input
#    register: result
#  - name: print variables
#    ansible.builtin.debug:
#      msg: Access Key {{ result.access_key }} and Secret Key {{ result.secret_key }}
#  - name: check outcomes
#    assert:
#      that:
#        - result.access_key
------------------------------------------------------------------------------------------

Uncomment the above 8 lines and run the playbook.

#        - result.secret_key
------------------------------------------------------------

### Actual Results

```console
access_key not found
secret_key not found
```

**Comment by carchi8py** [comment]
We have story (6551) open for this.

**Comment by carchi8py** [comment]
This should be fixed in the latest release 22.9.0

**Comment by vhulikal** [comment]
I will get back to you from my NetApp email. Thanks, Chris.

Sent from Yahoo Mail on Android 
 
  On Fri, Jan 26, 2024 at 1:34 PM, Chris ***@***.***> wrote:   


This should be fixed in the latest release 22.9.0
